conans/client/tools/files.py

```diff
@@ -374,3 +375,45 @@ def remove_files_by_mask(directory, pattern):
                 os.unlink(fullname)
                 removed_names.append(os.path.relpath(fullname, directory))
     return removed_names
+
+
+def fix_symlinks(conanfile, raise_if_error=False):
+    """ Fix the symlinks in the conanfile.package_folder: make symlinks relative and remove
+        those links to files outside the package (it will print an error, or raise
+        if 'raise_if_error' evaluates to true).
+    """
+    offending_files = []
+
+    def work_on_element(dirpath, element, token):
+        fullpath = os.path.join(dirpath, element)
+        if os.path.islink(fullpath):
+            link_target = os.readlink(fullpath)
+            link_abs_target = os.path.join(dirpath, link_target)
```

Comment:
is this correct thing to do? e.g.:
```
>>> os.path.join("C:\\Windows", "/dev/null")
'C:/dev/null'
```
returns broken result in case readlink returns `/dev/null`?

Reply:
Oh! Something to take into account: probably those _absolute_ links to `/dev/null` (Unix) and `...nul` is Windows is something we can identify and preserve the symlink. Do you know about any other _special_ file paths?

Also, for Windows, I'm missing one _if_ clause: `if os.path.isabs(link_rel_target)`

Reply:
Thinking about this issue. I feel like we need to detect the special Unix `/dev/null` file, but for Windows I assume you will [name the file `NUL`, `LPT1`, `COM1`,...](https://docs.microsoft.com/es-es/windows/win32/fileio/naming-a-file) instead of creating a symlink.

Reply:
there are number of such special files or pseudo-files, e.g. `/dev/zero`, `/dev/urandom`, `/dev/full`, etc. same on Windows with `LPT1`, `COM1`, `\\Device\\HarddiskVolume1\`, etc. maybe also we don't things in `/etc`, `/boot`, `/proc` to name a few. but in general, I think they are just absolute links outside of the package folder.